### PR TITLE
Opt: adapter V2 config for master slave split

### PR DESCRIPTION
**Pull request by Lvnszn**
**What this PR does**:
adapter V2 config for master slave split.
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/executor/redirect.go [hunk 1]
```diff
@@ -68,6 +68,35 @@ func NewRedirectExecutor(conf *config.Executor) proto.Executor {
 	return executor
 }
 
+func NewRedirectExecutorV2(conf *config.Executor, cluster *config.DataSourceCluster) proto.Executor {
+	executor := &RedirectExecutor{
+		mode:                conf.Mode,
+		preFilters:          make([]proto.PreFilter, 0),
```
Comment (AlexStocks) [inline pkg/executor/redirect.go:74]: delete the '0'

File: pkg/config/db.go [hunk 2]
```diff
@@ -200,3 +202,22 @@ func (t *DataSourceType) unmarshalText(text []byte) bool {
 	}
 	return true
 }
+
+var reg = regexp.MustCompile(`^r([0-9]*)w([0-9]*)$`)
```
Comment (jjeffcaii) [inline pkg/config/db.go:206]: Use`[0-9]+` intead?

Comment (Lvnszn) [reply pkg/config/db.go]: addressed

**Review by dongzl — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Review by jjeffcaii — APPROVED** [review]
LGTM

**Pull request merged**